### ISSUE TITLE
how to detect a file to be a python script

### ISSUE THREAD

**Issue by stefanseefeld** [spyder-ide/spyder#4643]
Is there a way to help spyder understand that a file (script) is in fact a python script, even if it doesn't use the usual '.py' extension and neither has the '#!' shebang magic ?
For emacs I'm embedding a `# -*- python -*-` line, but that obviously isn't recognized by spyder. Is there any other mechanism I could use ?

**Comment by ccordoba12** [comment]
No, sorry, there's no way to make Spyder understand that a file is actually a Python script.

What if we add a menu entry that lets users select a given syntax of a file, as many other editors do? I mean, something like this (taken from Sublime):

![seleccion_001](https://user-images.githubusercontent.com/365293/27517547-17b9126e-5994-11e7-93ad-50e9c3902576.png)

**Comment by stefanseefeld** [comment]
Why not add a (configurable) mechanism by which spyder would detect a file's type (i.e., syntax) ? I've taken my example above from emacs, but there are obviously many more editors that do something similar.
In any case, I agree that a mechanism to manually specify the syntax (including override any auto-detection) would be very useful.

**Comment by ccordoba12** [comment]
> there are obviously many more editors that do something similar.

The Emacs syntax is not a bad one, but I was not aware that other editors have a similar approach. Could you mention some other examples?

**Comment by stefanseefeld** [comment]
I know from experience that `vi` also detects the above syntax.
Many editors have mechanisms to define new file types, or at least let uses associate filename extensions with file types (and thus editor modes), including eclipse, atom, etc.
I'm not sure how many editors (other than emacs and vi) detect file types based on content.

**Comment by goanpeca** [comment]
We could use the pygments lexer detector?

**Comment by rlaverde** [comment]
The languages configuration (that we're planing to introduce in spyder4) could have a field "first line", that will be a regex to detect the file based in the first line (like vscode do)

`first_line = r"^#!/.*\\bpython[0-9.-]*\\b"`
